src/services/NodeService.ts

```diff
@@ -34,37 +49,68 @@ export class NodeService {
+            : _.uniqBy(profileNode, 'url').filter((n) => n.networkType === profile.networkType);
     }
 
-    public getNodes(profile: ProfileModel, repositoryFactory: RepositoryFactory, repositoryFactoryUrl: string): Observable<NodeModel[]> {
-        const storedNodes = this.getKnowNodesOnly(profile);
+    public async getNodes(
+        profile: ProfileModel,
+        repositoryFactory: RepositoryFactory,
+        repositoryFactoryUrl: string,
+        isOffline: boolean,
+    ): Promise<NodeModel[]> {
+        const storedNodes = await this.getKnowNodesOnly(profile, isOffline);
         const nodeRepository = repositoryFactory.createNodeRepository();
-
-        return nodeRepository.getNodeInfo().pipe(
-            map((dto: NodeInfo) =>
-                this.createNodeModel(
-                    repositoryFactoryUrl,
-                    profile.networkType,
-                    dto.friendlyName,
-                    undefined,
-                    dto.publicKey,
-                    dto.nodePublicKey,
+        return nodeRepository
+            .getNodeInfo()
+            .pipe(
+                map((dto: NodeInfo) =>
+                    this.createNodeModel(
+                        repositoryFactoryUrl,
+                        profile.networkType,
+                        dto.friendlyName,
+                        undefined,
+                        dto.publicKey,
+                        dto.nodePublicKey,
+                    ),
                 ),
-            ),
-            ObservableHelpers.defaultLast(this.createNodeModel(repositoryFactoryUrl, profile.networkType)),
-            map((currentNode) => _.uniqBy([currentNode, ...storedNodes], 'url')),
-            tap((p) => this.saveNodes(profile, p)),
-        );
+                ObservableHelpers.defaultLast(this.createNodeModel(repositoryFactoryUrl, profile.networkType)),
+                map((currentNode) => _.uniqBy([currentNode, ...storedNodes], 'url')),
+                tap((p) => this.saveNodes(profile, p)),
+            )
+            .toPromise();
     }
 
-    private loadStaticNodes(networkType: NetworkType): NodeModel[] {
-        return networkConfig[networkType].nodes.map((n) => {
-            return this.createNodeModel(n.url, networkType, n.friendlyName, true);
-        });
+    public async loadNodesFromStatisticService(networkType: NetworkType, isOffline?: boolean): Promise<NodeModel[]> {
+        const nodeSearchCriteria = {
+            nodeFilter: 'suggested',
+            limit: 30,
+        };
+        if (!isOffline && navigator.onLine) {
+            const data = await requestNodes(networkType, nodeSearchCriteria);
+            if (!data) {
+                return undefined;
+            }
+            return data.map((n) => {
+                // @ts-ignore
+                const isHttps = n.apiStatus?.isHttpsEnabled || false;
+                //@ts-ignore
+                const url = isHttps ? 'https://' + n.host + ':3001' : 'http://' + n.host + ':3000';
```

Comment:
There is new property `restGatewayUrl` you can use directly.
Check mobile wallet:
https://github.com/symbol/mobile-wallet/pull/246/files#diff-e8797ed74f24ca1eeb70998492cd0ee62dd5a33947a97d5ca928f773e0dc36f4R152